src/api/network_test.go

```diff
@@ -14,14 +14,41 @@ import (
 )
 
 func TestConnection(t *testing.T) {
+	bp := daemon.BlockchainProgress{
+		Current: 35,
+		Highest: 39,
+		Peers:   nil}
+	bp.Peers = append(bp.Peers, struct {
+		Address string `json:"address"`
+		Height  uint64 `json:"height"`
+	}{
+		Address: "127.3.5.1",
+		Height:  39,
+	})
+	bp.Peers = append(bp.Peers, struct {
+		Address string `json:"address"`
+		Height  uint64 `json:"height"`
+	}{
+		Address: "127.0.0.1",
+		Height:  12,
+	})
+	bp.Peers = append(bp.Peers, struct {
+		Address string `json:"address"`
+		Height  uint64 `json:"height"`
+	}{
+		Address: "127.0.5.1",
+		Height:  13,
+	})
+
 	tt := []struct {
-		name                       string
-		method                     string
-		status                     int
-		err                        string
-		addr                       string
-		gatewayGetConnectionResult *daemon.Connection
-		result                     *daemon.Connection
+		name                               string
+		method                             string
+		status                             int
+		err                                string
+		addr                               string
+		gatewayGetConnectionResult         *daemon.Connection
+		gatewayGetBlockchainProgressResult *daemon.BlockchainProgress
```

Comment:
should have test that mocks GetBlockchainProgress with an error

Reply:
done

Reply:
still need to write the original error message, not replaced with "GetBlockchainProgress failed"

Reply:
done